### PR TITLE
[#75] Schema.org mapping

### PR DESCRIPTION
**Pull request by metaodi**
Add schema.org mapping (see #75 for details)

This PR provides:
* Mapping from DCAT to schema.org/Dataset
* New parameter `profiles` to the catalog and dataset endpoints to be able to change the profiles when requesting the metadata (this allows to server DCAT and schema.org on the same instance), example `/dataset/my-dataset.jsonld?profiles=schemaorg`

### CODE REVIEW

File: ckanext/dcat/controllers.py [hunk 1]
```diff
@@ -32,10 +32,15 @@ def read_catalog(self, _format=None):
         if not _format:
             return HomeController().index()
 
+        _profiles = toolkit.request.params.get('profiles')
```
Comment (amercader) [inline ckanext/dcat/controllers.py]: Does it really make sense to ask for more than one profile at the time? Can you give an example? IIRC profiles used inheritance to extend them.

Comment (amercader) [reply ckanext/dcat/controllers.py]: Turns out I implemented it exactly like this at the lower level :upside_down_face: Ignore me!

**Comment by metaodi** [comment]
I found a way to check if a helper function is defined.

In https://github.com/ckan/ckan/pull/3042 an excpetion was added to the helper dictionary, so it's no longer possible to simply use `{% if h.structured_data %}` in jinja, as this triggers an exception. But fortunately jinja allows us to convert `h` to a normal dict, which in turns allows us to check if a key exists:

```
{% set helper_dict = dict(h) %}
{% if helper.dict.structured_data %}
{{ h.structured_data() }}
{% endif %}
```

**Comment by metaodi** [comment]
Unfortunately the approach I described above does not work in older versions of CKAN (<2.6), because earlier versions of jinja do not support the use of dict(). Therefore I added another template helper to check if a helper is available or not.

File: ckanext/dcat/utils.py [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
-from pylons import config
+from ckantoolkit import config, h
+
+try:
+    # CKAN >= 2.6
+    from ckan.exceptions import HelperError
+except ImportError:
+    # CKAN < 2.6
+    class HelperError(Exception):
+        pass
```
Comment (metaodi) [inline ckanext/dcat/utils.py]: I already added the `HelperError` to ckantoolkit to avoid this "hack" (https://github.com/ckan/ckantoolkit/pull/6)

**Comment by metaodi** [comment]
I rebased this PR to master to fix the merge conflicts.

**Comment by metaodi** [comment]
@amercader ping, can you review this PR?

**Comment by amercader** [comment]
sorry @metaodi, I'm a bit swamped right now but it's definitely on my list. Will try to review it this week.

File: README.md [hunk 3]
```diff
@@ -788,6 +799,88 @@ To disable this behavior, you can set the following config value in your ini fil
     ckanext.dcat.translate_keys = False
 
 
+## Structured data
+
+To add [structured data](https://developers.google.com/search/docs/guides/intro-structured-data) to dataset pages, activate the `structured_data` plugin in your ini file.
```
Comment (amercader) [inline README.md]: I took me quite a while to discover why I wasn't seeing the structured data json-ld after activating the `structured_data` plugin. Here we should mention that you need in `structured_data` in addition to the `dcat` plugin.

This basically means that if someone wants the schema.org support they also need to enable the links to RDF representation for datasets and catalog, as these are automatically added by the `dcat` plugin. This is probably fine, but we might want to add a config option to disable this for people that just want the schema.org support.

**Comment by amercader** [comment]
This looks good @metaodi . Only one minor change in the README and we can merge.

**Comment by metaodi** [comment]
@amercader I made the changes to the README and added the new config option for the RDF endpoints (+tests)

**Comment by amercader** [comment]
Thanks @metaodi, fantastic work as usual

**Comment by amercader** [comment]
This is worth a new release, I've released 0.0.7

**Pull request merged**